detekt-rules-errorprone/src/main/kotlin/io/gitlab/arturbosch/detekt/rules/bugs/MissingWhenCase.kt

```diff
@@ -74,40 +76,51 @@ class MissingWhenCase(config: Config = Config.empty) : Rule(config) {
         Debt.TWENTY_MINS
     )
 
+    private val allowElseExpression = valueOrDefault(ALLOW_ELSE_EXPRESSION, true)
+
     @Suppress("ReturnCount")
     override fun visitWhenExpression(expression: KtWhenExpression) {
         if (bindingContext == BindingContext.EMPTY) return
-        if (expression.elseExpression != null) return
+        if (allowElseExpression && expression.elseExpression != null) return
         if (expression.isUsedAsExpression(bindingContext)) return
         val subjectExpression = expression.subjectExpression ?: return
         val subjectType = subjectExpression.getType(bindingContext)
         val enumClassDescriptor = WhenChecker.getClassDescriptorOfTypeIfEnum(subjectType)
         if (enumClassDescriptor != null) {
-            val enumMissingCases = WhenChecker.getEnumMissingCases(expression, bindingContext, enumClassDescriptor)
-            if (enumMissingCases.isNotEmpty()) {
-                report(
-                    CodeSmell(
-                        issue, Entity.from(expression),
-                        "When expression is missing cases: ${enumMissingCases.joinToString()}. Either add missing " +
-                                "cases or a default `else` case."
-                    )
-                )
-            }
+            val enumMissingCases =
+                WhenChecker.getEnumMissingCases(expression, bindingContext, enumClassDescriptor)
+            getReport(enumMissingCases, expression)
         }
         val sealedClassDescriptor = WhenChecker.getClassDescriptorOfTypeIfSealed(subjectType)
         if (sealedClassDescriptor != null) {
             val sealedClassMissingCases =
                 WhenChecker.getSealedMissingCases(expression, bindingContext, sealedClassDescriptor)
-            if (sealedClassMissingCases.isNotEmpty()) {
-                report(
-                    CodeSmell(
-                        issue, Entity.from(expression),
-                        "When expression is missing cases: ${sealedClassMissingCases.joinToString()}. Either add " +
-                                "missing cases or a default `else` case."
-                    )
-                )
-            }
+            getReport(sealedClassMissingCases, expression)
         }
         super.visitWhenExpression(expression)
     }
+
+    private fun getReport(
```

Comment:
```suggestion
    private fun reportMissingCases(
```

I think the function name is a bit misleading as it's called `getReport` but is returning nothing (that's actually an idea for a Detekt rule 😁 )